### PR TITLE
 [WFCORE-2687] Addition of getCurrentOperationName() and getCurrentOperationParameter(String name) methods

### PR DESCRIPTION
**Pull request by lvydra**
https://issues.redhat.com/browse/WFCORE-2687

### CODE REVIEW

File: controller/src/main/java/org/jboss/as/controller/AbstractOperationContext.java [hunk 1]
```diff
@@ -541,6 +541,41 @@ public final String getCurrentAddressValue() {
...
+            } else {
+                return operation.get(parameterName);
+            }
+        } else {
+            throw new IllegalArgumentException(ControllerLogger.ROOT_LOGGER.invalidParameterName(parameterName));
+        }
+    }
+
+    private boolean isLegalParameterName(final String parameterName) {
+        return !(parameterName.equals(NAME) || parameterName.equals(OP) || parameterName.equals(OP_ADDR) || parameterName.equals(OPERATION_HEADERS));
```
Comment (bstansberry) [inline controller/src/main/java/org/jboss/as/controller/AbstractOperationContext.java:576]: @lvydra Please remove 'parameterName.equals(NAME) 'from this. My apologies; I should not have listed 'name' as an invalid key in the JIRA description. I've corrected the JIRA description.

File: controller/src/main/java/org/jboss/as/controller/logging/ControllerLogger.java [hunk 2]
```diff
@@ -3629,4 +3629,7 @@ OperationFailedRuntimeException capabilityAlreadyRegisteredInContext(String capa
 
     @Message(id = 476, value = "Value for attribute '%s' is invalid: either '%s' must be specified on its own or '%s' needs to be specified with at least one of '%s' or '%s'")
     OperationFailedException invalidCredentialReferenceValue(String attributeName, String clearTextAttributeName, String credentialStoreAttributeName, String clearTextAttributeName1, String alias);
+
+    @Message(id = 477, value = "Parameter name '%s' is invalid.")
+    OperationFailedException invalidParameterName(String parameterName);
```
Comment (bstansberry) [inline controller/src/main/java/org/jboss/as/controller/logging/ControllerLogger.java:3634]: The return type here should be IllegalArgumentException, as that is what you intend to throw.

File: controller/src/main/java/org/jboss/as/controller/AbstractOperationContext.java [hunk 3]
```diff
@@ -541,6 +541,41 @@ public final String getCurrentAddressValue() {
...
+            ModelNode operation = activeStep.operation;
+
+            assert operation != null;
+            if (!operation.has(parameterName) && nullable) {
+                return null;
+            } else {
+                return operation.get(parameterName);
+            }
+        } else {
+            throw new IllegalArgumentException(ControllerLogger.ROOT_LOGGER.invalidParameterName(parameterName));
```
Comment (bstansberry) [inline controller/src/main/java/org/jboss/as/controller/AbstractOperationContext.java:571]: ControllerLogger.ROOT_LOGGER.invalidParameterName should return IllegalArgumentException so you can just throw what it returns without wrapping.

File: controller/src/main/java/org/jboss/as/controller/OperationContext.java [hunk 4]
```diff
@@ -418,6 +418,30 @@
...
+     *
+     * @return operation name node
+     */
+    ModelNode getCurrentOperationName();
+
+    /**
+     * Get parameter node by its name
+     *
+     * @param name of desired parameter
+     * @return node for parameter of given name
```
Comment (bstansberry) [inline controller/src/main/java/org/jboss/as/controller/OperationContext.java:432]: This should also say 'or {@code null} if the operation has no such parameter'.

File: controller/src/main/java/org/jboss/as/controller/OperationContext.java [hunk 5]
```diff
@@ -418,6 +418,30 @@
      */
     String getCurrentAddressValue();
 
+    /**
+     * Get the node with current operation name
+     *
+     * @return operation name node
+     */
+    ModelNode getCurrentOperationName();
```
Comment (bstansberry) [inline controller/src/main/java/org/jboss/as/controller/OperationContext.java:426]: This should return String. The node will always hold a string.

File: controller/src/main/java/org/jboss/as/controller/OperationContext.java [hunk 6]
```diff
@@ -418,6 +418,30 @@
...
+     * @param name of desired parameter
+     * @return node for parameter of given name
+     */
+    ModelNode getCurrentOperationParameter(String name);
+
+    /**
+     * Get parameter node by its name
+     *
+     * @param name of desired parameter
+     * @param nullable whether return value can be null
```
Comment (bstansberry) [inline controller/src/main/java/org/jboss/as/controller/OperationContext.java:440]: whether the return value can be null if the operation has no parameter with the given name

File: controller/src/main/java/org/jboss/as/controller/OperationContext.java [hunk 7]
```diff
@@ -418,6 +418,30 @@
...
+     * @return node for parameter of given name
```
Comment (bstansberry) [inline controller/src/main/java/org/jboss/as/controller/OperationContext.java:441]: node for the parameter of the given name, or {@code null} if {@code nullable} is {@code true} and the operation has no parameter with the given name

**Review by bstansberry — CHANGES_REQUESTED** [review]

**Comment by lvydra** [comment]
Hi @bstansberry , thank you for your review. I have added requested changes.

**Review by bstansberry — APPROVED** [review]

**Comment by bstansberry** [comment]
Thanks, @lvydra !

**Pull request merged**